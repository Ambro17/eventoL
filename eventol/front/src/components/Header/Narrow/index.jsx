import React from 'react'
import PropTypes from 'prop-types'
import Logo from '../../../components/Logo';
import NavigationNarrow from '../../../components/Navigation/Narrow';
import UserProfileNarrow from '../../../components/UserProfile/Narrow';
import SignInNarrow from '../../../components/SignIn/Narrow';
import './index.scss';


export default class HeaderNarrow extends React.PureComponent {
  static propTypes = {
    logoHeader: PropTypes.string,
    user: PropTypes.object
  };

  handleToggle(){
    const linksEl = document.querySelector('.narrow-links');
    const display = linksEl.style.display;
    linksEl.style.display = (display === 'block') ? 'none' : 'block';
  }

  render(){
    const {user, logoHeader} = this.props;
    return (
      <div className='nav-narrow'>
        <Logo logoHeader={logoHeader} />
        <div className='nav-right'>
          <a href='#' onClick={this.handleToggle}>
            <i className='fa fa-bars fa-2x' />
          </a>
          <div className='narrow-links'>
            <NavigationNarrow />
            {user && <UserProfileNarrow user={user} />}
            {!user && <SignInNarrow />}
          </div>
        </div>
      </div>
    )
  }
}
