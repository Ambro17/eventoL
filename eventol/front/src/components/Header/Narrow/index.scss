.navNarrow {
  display: flex;
  padding: 40px;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  box-sizing: border-box;

  .navRight {
    margin-left: auto;
    i {
      float: right;
      z-index: 5;
    }
    .narrowLinks {
      display: none;
      a {
        display: block;
        float: right;
        clear: right;
      }
    }
  }
}

@media (min-width: 950px) {
  .navNarrow {
    display: none;
  }
}

a {
  color: white;
}
