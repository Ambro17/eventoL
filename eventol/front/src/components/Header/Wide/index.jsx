import React from 'react'
import PropTypes from 'prop-types'
import Logo from '../../../components/Logo'
import Navigation from '../../../components/Navigation'
import UserProfile from '../../../components/UserProfile'
import SignIn from '../../../components/SignIn'

import './index.scss'


const HeaderWide = ({user, logoHeader}) => (
    <Logo logoHeader={logoHeader}/>
  <div className='nav-wide'>
    <Navigation />
    {user && <UserProfile user={user}/>}
    {!user && <SignIn/>}
  </div>
);

HeaderWide.propTypes = {
  user: PropTypes.object,
  logoHeader: PropTypes.string
};

export default HeaderWide;
