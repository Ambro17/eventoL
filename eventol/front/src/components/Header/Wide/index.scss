.navWide {
  display: none;
}

@media (min-width: 750px) {
  .navWide {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    width: 100vw;
    box-sizing: border-box;
  }
}
