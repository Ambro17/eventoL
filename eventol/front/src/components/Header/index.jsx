import React from 'react'
import PropTypes from 'prop-types'
import HeaderNarrow from './Narrow';
import HeaderWide from './Wide';
import './index.scss';

const Header = ({user, logoHeader}) => (
  <header className='header'>
    <HeaderWide logoHeader={logoHeader} user={user} />
    <HeaderNarrow logoHeader={logoHeader} user={user} />
  </header>
);

Header.propTypes = {
  logoHeader: PropTypes.string,
  user: PropTypes.object
};

export default Header;
