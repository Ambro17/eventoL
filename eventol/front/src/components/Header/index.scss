@import '../../styles/variables';
@import '../../styles/commons';

.header {
  @extend %flex-element;
  // FF3.6-15
  background: -webkit-linear-gradient(top, $gradient-color 0%, transparent 100%);

  // Chrome10-25, Safari5.1-6
  background: linear-gradient(to bottom, $gradient-color 0%, transparent 100%);

  box-sizing: border-box;

  // W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#a6000000', endColorstr='#00000000', GradientType=0);

  left: 0;
  padding: 40px;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: 5;
}
