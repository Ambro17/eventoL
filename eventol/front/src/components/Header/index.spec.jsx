import React from 'react';
import renderer from 'react-test-renderer';

import Header from '.';

describe('Header', () => {
  let component, tree, user, logo, isMobile;

  beforeEach(() => {
    /* eslint-disable camelcase */
    user = {
      first_name: 'first_name',
      last_name: 'last_name',
    };
    /* eslint-enable camelcase */
    logo = 'eventol.png';
  });

  describe('Mobile version', () => {
    beforeEach(() => {
      isMobile = true;
      component = renderer.create(
        <Header logoHeader={logo} user={user} isMobile={isMobile} />,
      );
      tree = component.toJSON();
    });

    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Desktop version', () => {
    beforeEach(() => {
      isMobile = false;
      component = renderer.create(
        <Header logoHeader={logo} user={user} isMobile={isMobile} />,
      );
      tree = component.toJSON();
    });

    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
