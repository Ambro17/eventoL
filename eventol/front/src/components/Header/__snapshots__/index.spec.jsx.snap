// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Desktop version Snapshot 1`] = `
<header
  className="header"
>
  <div
    className="nav-wide"
  >
    <div
      className="logo-component"
      id="logo"
    >
      <a
        href="/"
      >
        <img
          src="eventol.png"
        />
      </a>
    </div>
    <div
      className="navigation"
      id="navigation"
    >
      <nav>
        <ul>
          <a
            onClick={[Function]}
            onKeyPress={[Function]}
            role="link"
            tabIndex="0"
          >
            <li>
              Search
            </li>
          </a>
          <a
            href="http://eventol.github.io/eventoL/#/"
          >
            <li>
              Documentation
            </li>
          </a>
          <a
            href="#featured"
          >
            <li>
              Featured
            </li>
          </a>
          <a
            href="#recent"
          >
            <li>
              Recent
            </li>
          </a>
          <a
            href="/report"
          >
            <li>
              Generic report
            </li>
          </a>
        </ul>
      </nav>
    </div>
    <div
      className="user-profile"
    >
      <a
        href="/accounts/profile/"
      >
        <div
          className="user"
        >
          <div
            className="name"
          >
            first_name last_name
          </div>
          <div
            className="image"
          >
            <div
              className="fa fa-user fa-2x"
            />
          </div>
        </div>
      </a>
    </div>
  </div>
</header>
`;

exports[`Header Mobile version Snapshot 1`] = `
<header
  className="header"
>
  <div
    className="nav-narrow"
  >
    <div
      className="logo-component"
      id="logo"
    >
      <a
        href="/"
      >
        <img
          src="eventol.png"
        />
      </a>
    </div>
    <div
      className="nav-right"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <i
          className="fa fa-bars fa-2x"
        />
      </a>
      <div
        className="narrow-links"
      >
        <div
          className="navigation"
          id="navigation"
        >
          <nav>
            <ul>
              <a
                onClick={[Function]}
                onKeyPress={[Function]}
                role="link"
                tabIndex="0"
              >
                <li>
                  Search
                </li>
              </a>
              <a
                href="http://eventol.github.io/eventoL/#/"
              >
                <li>
                  Documentation
                </li>
              </a>
              <a
                href="#featured"
              >
                <li>
                  Featured
                </li>
              </a>
              <a
                href="#recent"
              >
                <li>
                  Recent
                </li>
              </a>
              <a
                href="/report"
              >
                <li>
                  Generic report
                </li>
              </a>
            </ul>
          </nav>
        </div>
        <div
          className="user-profile"
        >
          <a
            href="/accounts/profile/"
          >
            <div
              className="user"
            >
              <div
                className="name"
              >
                first_name last_name
              </div>
              <div
                className="image"
              >
                <div
                  className="fa fa-user fa-2x"
                />
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</header>
`;
